werkzeug/wrappers.py

```diff
@@ -1443,12 +1484,43 @@ def make_conditional(self, request_or_environ):
             # wsgiref.
             if 'date' not in self.headers:
                 self.headers['Date'] = http_date()
+            if accept_ranges is True:
+                accept_ranges = "bytes"
+            elif accept_ranges is None:
+                accept_ranges = "none"
+            elif not isinstance(accept_ranges, string_types):
+                accept_ranges = None
```

Comment:
By the docstring, strings are also allowed. My point is that `accept_ranges` will be silently set to `None` if I e.g. pass something like `object()`, `42`, etc.

Also IMO we should convert any string type to the `str` type (use `werkzeug._compat.to_native`).